autoscheduler/user_sessions/utils/retrieve_data_session.py

```diff
@@ -0,0 +1,46 @@
+from contextlib import contextmanager
+from django.contrib.sessions.backends.db import SessionStore
+from django.contrib.sessions.models import Session
+from user_sessions.models import UserToDataSession
+
+@contextmanager
+def retrieve_data_session(request):
+    """ Checks if the user already has a session assigned to them in which
+        to store their data. If so, saves the session id of that session to
+        their current session. Otherwise, creates a session for them, and a
+        user_to_data_session entry that associates them with the session
+        before saving that session id to their current session
+    """
+    # Get user
+    user = request.user
+    user_id = user.id
```

Comment:
```suggestion
    user_id = request.user.id
```
One more simplifaction and this should be ready to merge, thought I pointed it out in the last round of review but I forgot to oops